### DISCUSSION TITLE
Mac M1 GPU support

### DISCUSSION THREAD

**Discussion by mattiasurricchioprima** [speechbrain/speechbrain#1791] [Q&A] [answered]
Hi everyone, I was wondering if Speechbrain supports the new M1 GPU with MPS.

This is my very simple code for a speech separation task:

```
from speechbrain.pretrained.interfaces import SepformerSeparation
import torchaudio

separator = SepformerSeparation.from_hparams(source="speechbrain/sepformer-wsj02mix", savedir="./pretrained-sepformer-wsj02mix",  run_opts={"device": "mps"})

s1, fs = torchaudio.load('./Italian_Conversational_Speech_Corpus.wav')
resampler = torchaudio.transforms.Resample(fs, 8000)
s1 = resampler(s1)

est_sources = separator.separate_batch(s1)
```

However im getting this error: `ValueError: invalid type: 'torch.mps.FloatTensor'`.

The M1 GPU is correctly identified and available in pytorch: running `torch.backends.mps.is_available()` returns `True` and I have also been able to run some PyTorch models on my M1 GPU, so the GPU itself is available and working.

Is Speechbrain supporting M1 GPU? If yes, what am I missing? Probably I should send the data `s1` to my GPU with the `to_device()` method? But I think this is not the case because in other tutorial/docs where they use CUDA, the data transfer to the GPU is embedded in the library (at least this is what it looks like).

Any help?

**Comment by pplantinga** [comment]
See relevant issue: https://github.com/speechbrain/speechbrain/issues/1794, seems this is a PyTorch bug that is being worked on.

**Comment by Adel-Moumen** [comment, marked as answer]
#1805
